modules/RTC/TraceablePeerConnection.js

```diff
@@ -2289,6 +2296,31 @@ TraceablePeerConnection.prototype._mungeOpus = function(description) {
     });
 };
 
+/**
+ * Sets up the _dtlsTransport object and initializes callbacks for it.
+ */
+TraceablePeerConnection.prototype._initializeDtlsTransport = function() {
+    // We are assuming here that we only have one bundled transport here
+    if (this._dtlsTransport) {
+        return;
+    }
+
+    const senders = this.peerconnection.getSenders();
+
+    if (senders.length !== 0 && senders[0].transport) {
+        this._dtlsTransport = senders[0].transport;
+
+        this._dtlsTransport.onerror = error => {
+            logger.error(`${this} DtlsTransport error: ${error}`);
+        };
+
+        this._dtlsTransport.onstatechange = () => {
+            this.trace('dtlsTransport.onstatechange', this._dtlsTransport.state);
```

Comment:
Do we need to handle the cases when the DTLS transport state changes to `failed` ? Does the ICE connection state also change when that happens ? If that is the case, we are good but otherwise I am wondering the client might end up in a broken state where it expects media to be sent and received. Do we need to attempt to reload the client in those cases ?

Reply:
So if the ICE succeeded and DTLS fails, ICE will stay in the success stated. In other words: a DTLS failure doesn't result in any other state changes.

A very good point. Yeah we could/should attempt to do something if we encounter a failure. I would propose that I'll do an action part as a follow up to this. Because honestly I'm not sure how the failures will look like. I have not tried to test a failure locally so far.
I would think the `onerror` callback is where we would want to hook up any actions.